pandas/io/excel/_openpyxl.py

```diff
@@ -412,7 +419,25 @@ def write_cells(
         _style_cache: Dict[str, Dict[str, Serialisable]] = {}
 
         if sheet_name in self.sheets:
-            wks = self.sheets[sheet_name]
+            if "r+" in self.mode:
+                if self.if_sheet_exists == "new":
+                    wks = self.book.create_sheet()
+                    # openpyxl will create a name for the new sheet by appending digits
+                    wks.title = sheet_name
+                    self.sheets[wks.title] = wks
+                elif self.if_sheet_exists == "replace":
+                    wks = self.sheets[sheet_name]
+                    wks.delete_cols(1, wks.max_column)
+                elif self.if_sheet_exists == "overwrite":
+                    wks = self.sheets[sheet_name]
+                elif self.if_sheet_exists == "fail":
+                    raise ValueError(f"Sheet '{sheet_name}' already exists.")
+                else:
+                    raise ValueError(
+                        f"'{self.if_sheet_exists}' is not valid for if_sheet_exists"
+                    )
+            else:
+                wks = self.sheets[sheet_name]
```

Comment:
This is hit if we write to the same sheet multiple times with the same write mode writer. E.g.

```
import pandas as pd
import io

df1 = pd.DataFrame({
    "col1": ["apple", "banana"],
    "col2": [1, 2],
})


excel = io.BytesIO()

with pd.ExcelWriter(excel, engine="openpyxl", mode="w") as writer:
    # write two sheets
    df1.to_excel(writer, sheet_name="sheet1", index=False)
    df1.to_excel(writer, sheet_name="sheet1", index=False, startcol=3)

df = pd.read_excel(excel)
print(df)

#      col1  col2  Unnamed: 2  col1.1  col2.1
# 0   apple     1         NaN   apple       1
# 1  banana     2         NaN  banana       2
```

This is the same across all of the engines and I don't think there is any reason to change it. This situation doesn't need the same set of options because they can easily be replicated in user code.

It doesn't look like there are any tests for this though, for any of the engines. Should I add one to `test_writers`?

Reply:
Thanks - makes sense. Yes, a test would be great.

Reply:
Created a separate issue to